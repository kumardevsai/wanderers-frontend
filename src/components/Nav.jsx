import React, { Component } from 'react';
import { observer, inject } from 'mobx-react';
// import { Link } from 'react-router-dom';
import {
  NavWrapper,
  Navigation,
  List,
  ListItem,
  NavLink
} from '../elements/nav';

@inject('WanderersStore')
@observer
export default class Nav extends Component {
  render() {
    const { WanderersStore } = this.props;
    const user = WanderersStore.user;

    return (
      <NavWrapper>
        <Navigation>
          {user ? (
            <List>
              <ListItem>
                <NavLink to="/">HOME</NavLink>
              </ListItem>
              <ListItem>
                <NavLink to="/trips/new">PLAN TRIP</NavLink>
              </ListItem>
              <ListItem>
                <NavLink to="/trips">MY TRIPS</NavLink>
              </ListItem>
              <ListItem>
                <NavLink to="/places">MAP</NavLink>
              </ListItem>
              <ListItem>
                <NavLink to="/logout">LOG OUT</NavLink>
              </ListItem>
            </List>
          ) : (
            <List>
              <ListItem>
                <NavLink to="/">HOME</NavLink>
              </ListItem>
              <ListItem>
                <NavLink to="/signup">SIGN UP</NavLink>
              </ListItem>
              <ListItem>
                <NavLink to="/login">LOGIN</NavLink>
              </ListItem>
            </List>
          )}
        </Navigation>
      </NavWrapper>
    );
  }
}
