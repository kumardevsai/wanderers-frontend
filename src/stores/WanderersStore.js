import { observable, action } from 'mobx';
import AuthApi from '../services/AuthApi';
import PlacesApi from '../services/PlacesApi';

class WanderersStore {
  @observable user = {};

  @observable
  viewport = {
    width: window.innerWidth,
    height: window.innerHeight,
    latitude: 43.65323,
    longitude: -79.38318,
    zoom: 12
  };

  @observable places = [];

  constructor() {
    this.authApi = new AuthApi();
    this.placesApi = new PlacesApi();

    // ensure that there is a user & load user into state (makes it available in store)
    const session_user = sessionStorage.getItem('user');
    if (session_user) {
      this.user = JSON.parse(session_user);
    }
  }

  @action
  login = async (email, password) => {
    const response = await this.authApi.login(email, password);

    this.user = response.data;
    sessionStorage.setItem('user', JSON.stringify(this.user));
  };

  @action
  signup = async (name, email, password, image) => {
    const response = await this.authApi.signup(name, email, password, image);

    this.user = response.data;
    sessionStorage.setItem('user', JSON.stringify(this.user));
  };

  @action
  searchPlaces = async (lat, lon) => {
    const response = await this.placesApi.search('abc', lat, lon);

    this.places = response.data;
  };
}

const singleton = new WanderersStore();
export default singleton;
