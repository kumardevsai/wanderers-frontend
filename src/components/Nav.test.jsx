/*global React, render*/

import Nav from './Nav';
import { StaticRouter } from 'react-router-dom';
import { Provider, useStaticRendering } from 'mobx-react';
import WanderersStore from '../stores/WanderersStore';

useStaticRendering(true);

it('renders nav for authenticated user', () => {
  WanderersStore.user = true;

  const nav = render(
    <Provider WanderersStore={WanderersStore}>
      <StaticRouter context={{}}>
        <Nav />
      </StaticRouter>
    </Provider>
  );
  expect(nav).toMatchSnapshot();
});

it('renders nav for unauthenticated user', () => {
  WanderersStore.user = false;

  const nav = render(
    <Provider WanderersStore={WanderersStore}>
      <StaticRouter context={{}}>
        <Nav />
      </StaticRouter>
    </Provider>
  );
  expect(nav).toMatchSnapshot();
});
