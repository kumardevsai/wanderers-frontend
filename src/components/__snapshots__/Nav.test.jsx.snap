// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nav for authenticated user 1`] = `
<div
  class="sc-bdVaJa hlxeFr"
>
  <nav
    class="sc-bwzfXH efklKu"
  >
    <ul
      class="sc-htpNat gydiuv"
    >
      <li
        class="sc-bxivhb dWbHYm"
      >
        <a
          class="sc-ifAKCX iFVAIq"
          href="/"
        >
          HOME
        </a>
      </li>
      <li
        class="sc-bxivhb dWbHYm"
      >
        <a
          class="sc-ifAKCX iFVAIq"
          href="/trips/new"
        >
          PLAN TRIP
        </a>
      </li>
      <li
        class="sc-bxivhb dWbHYm"
      >
        <a
          class="sc-ifAKCX iFVAIq"
          href="/trips"
        >
          MY TRIPS
        </a>
      </li>
      <li
        class="sc-bxivhb dWbHYm"
      >
        <a
          class="sc-ifAKCX iFVAIq"
          href="/places"
        >
          MAP
        </a>
      </li>
      <li
        class="sc-bxivhb dWbHYm"
      >
        <a
          class="sc-ifAKCX iFVAIq"
          href="/logout"
        >
          LOG OUT
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`renders nav for unauthenticated user 1`] = `
<div
  class="sc-bdVaJa hlxeFr"
>
  <nav
    class="sc-bwzfXH efklKu"
  >
    <ul
      class="sc-htpNat gydiuv"
    >
      <li
        class="sc-bxivhb dWbHYm"
      >
        <a
          class="sc-ifAKCX iFVAIq"
          href="/"
        >
          HOME
        </a>
      </li>
      <li
        class="sc-bxivhb dWbHYm"
      >
        <a
          class="sc-ifAKCX iFVAIq"
          href="/signup"
        >
          SIGN UP
        </a>
      </li>
      <li
        class="sc-bxivhb dWbHYm"
      >
        <a
          class="sc-ifAKCX iFVAIq"
          href="/login"
        >
          LOGIN
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
